entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -45,43 +77,59 @@ bool DirectionalGaussianBlurFilterContents::RenderFilter(
 
   auto& host_buffer = pass.GetTransientsBuffer();
 
-  // Because this filter is intended to be used with only one input parameter,
-  // and GetBounds just increases the input size by a factor of the direction,
-  // we we can just scale up the UVs by the same amount and don't need to worry
-  // about mapping the UVs to the destination rect (like we do in
-  // BlendFilterContents).
-
   auto input = inputs[0]->GetSnapshot(renderer, entity);
-  auto input_size = input->texture->GetSize();
+  auto input_bounds = inputs[0]->GetBounds(entity);
+  auto filter_bounds = GetBounds(entity);
 
   auto transformed_blur =
       entity.GetTransformation().TransformDirection(blur_vector_);
 
-  auto uv_offset = transformed_blur.Abs() / input_size;
   // LTRB
   Scalar uv[4] = {
-      -uv_offset.x,
-      -uv_offset.y,
-      1 + uv_offset.x,
-      1 + uv_offset.y,
+      (filter_bounds.GetLeft() - input_bounds.GetLeft()) /
+          input_bounds.size.width,
+      (filter_bounds.GetTop() - input_bounds.GetTop()) /
+          input_bounds.size.height,
+      1 + (filter_bounds.GetRight() - input_bounds.GetRight()) /
+              input_bounds.size.width,
+      1 + (filter_bounds.GetBottom() - input_bounds.GetBottom()) /
+              input_bounds.size.height,
+  };
+
+  auto source = source_override_ ? source_override_ : inputs[0];
+  auto source_texture = source->GetSnapshot(renderer, entity);
+  auto source_bounds = source->GetBounds(entity);
+
+  // LTRB
+  Scalar uv_src[4] = {
+      (filter_bounds.GetLeft() - source_bounds.GetLeft()) /
+          source_bounds.size.width,
+      (filter_bounds.GetTop() - source_bounds.GetTop()) /
+          source_bounds.size.height,
+      1 + (filter_bounds.GetRight() - source_bounds.GetRight()) /
+              source_bounds.size.width,
+      1 + (filter_bounds.GetBottom() - source_bounds.GetBottom()) /
+              source_bounds.size.height,
```

Comment:
I'm not clear on how these calculations work in a rotated coordinate system, but I don't think we'll encounter those in the near term anyway.

Reply:
FWIW, this will be replaced when we add a [local transform](https://github.com/flutter/impeller/pull/103#discussion_r837987229) for filter inputs.